code/game/objects/items/devices/pin_extractor.dm

```diff
@@ -0,0 +1,51 @@
+/obj/item/device/pin_extractor
+	name = "firing pin extractor"
+	desc = "A device that is capable of removing firing pin without damaging it."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "pin_extractor"
+	item_state = "pin_extractor"
+	w_class = 2
+	flags = CONDUCT
+	action_button_name = "Toggle extractor"
+	var/on = 0
+	var/activation_sound = 'sound/effects/lighton.ogg'
+	origin_tech = list(TECH_MATERIAL = 3, TECH_ENGINEERING = 3, TECH_MAGNET = 4)
+
+
+/obj/item/device/pin_extractor/update_icon()
+	if(on)
+		icon_state = "[initial(icon_state)]-on"
+	else
+		icon_state = "[initial(icon_state)]"
+
+/obj/item/device/pin_extractor/attack_self(mob/user)
+	on = !on
+	if(on && activation_sound)
```

Comment:
What is the purpose of having to turn it on, before you can use it ?
What prevents you from leaving it on at all times ?

Reply:
Nothing, just a thing to have animation working.